org-roam.el

```diff
@@ -253,17 +253,50 @@ it as FILE-PATH."
                       link-type
                       (list :content content :point begin)))))))))
 
-(defun org-roam--extract-titles ()
+(defcustom org-roam-title-include-subdirs nil
+  "When non-nil, include sub-directories relative to
+`org-roam-directory' in title completions."
+  :type 'boolean
+  :group 'org-roam)
+
+(defcustom org-roam-title-subdir-separator "/"
+  "String to use to separate subdirs when
+`org-roam-title-include-subdirs' is non-nil."
+  :type 'string
+  :group 'org-roam)
```

Comment:
Can we use some sort of format-string instead, like we already do so with `org-roam-link-title-format`? So given `org-roam-directory/biblio/title.org` and `"(%s) %s"` (or `"(=subdir=) =title="`) set as a format-string it would yield to `(biblio) Title`.

It can be more complicated, but would give users more flexibility (e.g. placing subdirectory at the end of the string).

Reply:
That's a good idea.  I'll see what I can do.

Reply:
@zaeph maybe instead of the formatting string it would be simpler to preset this to a function that would take subdirectory and title as a 2 separate arguments. The return value would yield the final title. This approach is even more flexible (user can decide whether format something and how) and might be more simpler.

Reply:
Sounds good.  I like the fact that we're allowing most of our switches to accept functions for extra customisation.

Reply:
Let me finish wrapping my head around `org-roam-backlink` and I'll have a look.

Reply:
Implemented in cb1b2c01cbfbc20308da00e5316aa01c3bb8dfc6.